src/image_gen/http_image_gen_calculator.cc

```diff
@@ -0,0 +1,130 @@
+
+public:
+    static absl::Status GetContract(CalculatorContract* cc) {
+        RET_CHECK(!cc->Inputs().GetTags().empty());
+        RET_CHECK(!cc->Outputs().GetTags().empty());
+        cc->Inputs().Tag(INPUT_TAG_NAME).Set<ovms::HttpPayload>();
+        cc->InputSidePackets().Tag(IMAGE_GEN_SESSION_SIDE_PACKET_TAG).Set<ImageGenerationPipelinesMap>();  // TODO: template?
+        cc->Outputs().Tag(OUTPUT_TAG_NAME).Set<std::string>();
+        return absl::OkStatus();
+    }
+
+    absl::Status Close(CalculatorContext* cc) final {
+        SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "ImageGenCalculator [Node: {} ] Close", cc->NodeName());
+        return absl::OkStatus();
+    }
+
+    absl::Status Open(CalculatorContext* cc) final {
+        SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "ImageGenCalculator  [Node: {}] Open start", cc->NodeName());
+        return absl::OkStatus();
+    }
+
+    absl::Status Process(CalculatorContext* cc) final {
+        SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "ImageGenCalculator  [Node: {}] Process start", cc->NodeName());
+
+        ImageGenerationPipelinesMap pipelinesNap = cc->InputSidePackets().Tag(IMAGE_GEN_SESSION_SIDE_PACKET_TAG).Get<ImageGenerationPipelinesMap>();
+        auto it = pipelinesNap.find(cc->NodeName());
+        RET_CHECK(it != pipelinesNap.end()) << "Could not find initialized Image Gen node named: " << cc->NodeName();
+        auto pipe = it->second;
+
+        auto payload = cc->Inputs().Tag(INPUT_TAG_NAME).Get<ovms::HttpPayload>();
+        if (payload.parsedJson->HasParseError())
+            return absl::InvalidArgumentError("Failed to parse JSON");
+
+        if (!payload.parsedJson->IsObject()) {
+            return absl::InvalidArgumentError("JSON body must be an object");
+        }
+        SET_OR_RETURN(std::string, prompt, getPromptField(payload));
+
+        // TODO: Support more pipeline types
+        // Depending on URI, select text2ImagePipeline/image2ImagePipeline/inpaintingPipeline
+
+        // curl -X POST localhost:11338/v3/images/generations -H "Content-Type: application/json" -d '{ "model": "endpoint", "prompt": "A cute baby sea otter", "n": 1, "size": "1024x1024" }'
+        // FIXME routing request to different pipelines (enum?)
+        ov::genai::Text2ImagePipeline request = pipe->text2ImagePipeline.clone();
+        SET_OR_RETURN(ov::AnyMap, requestOptions, getImageGenerationRequestOptions(payload, pipe->args));
+        // preview limitation put here to not mess up tests underneath
+        auto imagesPerPromptIt = requestOptions.find("num_images_per_prompt");
+        if (imagesPerPromptIt != requestOptions.end()) {
+            auto numImages = imagesPerPromptIt->second.as<int>();
+            if (numImages != 1) {
+                return absl::InvalidArgumentError(absl::StrCat("In 2025.2 only 1 image in response is supported but requested:", numImages));
+            }
+        }
+        std::unique_ptr<ov::Tensor> images;
```

Comment:
ov::Tensor is a wrapper to shared ptr. Copying ov::Tensor does not copy memory underneath. Why cant we just operate on ov::Tensor?

Reply:
I think it has to do with try-catch and getting out of scope without destroying ov::Tensor?